src/tools/run-make-support/src/symbols.rs

```diff
@@ -42,3 +42,46 @@ pub fn any_symbol_contains(path: impl AsRef<Path>, substrings: &[&str]) -> bool
         false
     })
 }
+
+/// Get a list of symbols that are in `symbol_names` but not the final binary.
+///
+/// The symbols must also match `pred`.
+///
+/// The symbol names must match exactly.
+///
+/// Panics if `path` is not a valid object file readable by the current user.
+pub fn missing_exact_symbols<'a>(
+    path: impl AsRef<Path>,
+    symbol_names: &[&'a str],
+    pred: impl Fn(&Symbol<'_, '_>) -> bool,
+) -> Vec<&'a str> {
+    let mut found = vec![false; symbol_names.len()];
+    with_symbol_iter(path, |syms| {
+        for sym in syms.filter(&pred) {
+            for (i, symbol_name) in symbol_names.iter().enumerate() {
+                found[i] |= sym.name_bytes().unwrap() == symbol_name.as_bytes();
+            }
+        }
+    });
+    return found
+        .iter()
+        .enumerate()
+        .filter_map(|(i, found)| if !*found { Some(symbol_names[i]) } else { None })
+        .collect();
+}
+
+/// Assert that the symbol file contains all of the listed symbols and they all match the given predicate
```

Comment:
Suggestion: ditto above, we should describe on macOS we expect `_` prefix in addition to the non-underscore-prefixed symbol name.